pkg/sass-parser/lib/src/statement/forward-rule.test.ts

```diff
@@ -0,0 +1,930 @@
+          let clone: ForwardRule;
+          beforeEach(() => {
+            clone = original.clone({show: undefined});
+          });
+
+          it('changes show', () => expect(clone.show).toBeUndefined());
+
+          it('changes params', () =>
+            expect(clone.params).toBe('"foo" as bar-*  with ($zip: "zap")'));
+        });
+      });
+
+      describe('hide', () => {
+        describe('defined', () => {
+          let clone: ForwardRule;
+          beforeEach(() => {
+            clone = original.clone({hide: {variables: ['flip']}});
+          });
+
+          it('changes show', () => expect(clone.show).toBeUndefined());
+
+          it('changes hide', () =>
+            expect(clone.hide).toEqual({
+              mixinsAndFunctions: new Set([]),
+              variables: new Set(['flip']),
+            }));
+
+          it('changes params', () =>
+            expect(clone.params).toBe(
+              '"foo" as bar-* hide $flip  with ($zip: "zap")',
+            ));
+        });
+
+        describe('undefined', () => {
+          let clone: ForwardRule;
+          beforeEach(() => {
+            clone = original.clone({hide: undefined});
+          });
+
+          it('preserves show', () =>
+            expect(clone.show).toEqual({
+              mixinsAndFunctions: new Set(['baz']),
+              variables: new Set(['qux']),
+            }));
+
+          it('preserves hide', () => expect(clone.hide).toBeUndefined());
+
+          it('preserves params', () =>
+            expect(clone.params).toBe(
+              '"foo" as bar-* show baz, $qux  with ($zip: "zap")',
+            ));
+        });
+      });
+
+      describe('configuration', () => {
+        describe('defined', () => {
+          let clone: ForwardRule;
+          beforeEach(() => {
+            clone = original.clone({configuration: new Configuration()});
+          });
+
+          it('changes configuration', () =>
+            expect(clone.configuration.size).toBe(0));
+
+          it('changes params', () =>
+            expect(clone.params).toBe('"foo" as bar-* show baz, $qux'));
+        });
+
+        describe('undefined', () => {
+          let clone: ForwardRule;
+          beforeEach(() => {
+            clone = original.clone({configuration: undefined});
+          });
+
+          it('preserves configuration', () => {
+            expect(clone.configuration.size).toBe(1);
+            expect(clone.configuration.parent).toBe(clone);
+            const variables = [...clone.configuration.variables()];
+            expect(variables[0].variableName).toBe('zip');
+            expect(variables[0]).toHaveStringExpression('expression', 'zap');
+          });
+
+          it('preserves params', () =>
+            expect(clone.params).toBe(
+              '"foo" as bar-* show baz, $qux  with ($zip: "zap")',
+            ));
+        });
+      });
+    });
+  });
+
+  // Can't JSON-serialize this until we implement Configuration.source.span
```

Comment:
Is there a github issue we can link to here?

Reply:
It's really just part of the general task of implementing `sass-parser`.